kolibri/content/api.py

```diff
@@ -122,12 +122,16 @@ def filter_next_steps(self, queryset, value):
 
         completed_content_nodes = queryset.filter(content_id__in=completed_content_ids).order_by()
 
-        return queryset.exclude(pk__in=completed_content_nodes).filter(
+        # Filter to only show content that the user has not engaged in, so as not to be redundant with resume
+        engaged_content_nodes = queryset.filter(content_id__in=ContentSummaryLog.objects.filter(
+            user=value).values_list('content_id', flat=True))
```

Comment:
Can this not be `queryset.filter(contentsummarylog_set__user=value)`? Does the reverse relation have an identifier?

Reply:
ContentSummaryLog does not FK onto ContentNodes (and `content_id` is not unique across ContentNodes), so that's why this query is not great.